doc/rst/usingchapel/editors/vim.rst

```diff
@@ -0,0 +1,132 @@
+      filetype plugin indent on  " enable filetype detection, plugins, indent
+      set autoindent             " turn on auto indenting
+
+   Other useful commands are available in the
+   ``highlight/vim/sample-vimrc`` file.
+
+Vim-Plug Installation:
+^^^^^^^^^^^^^^^^^^^^^^
+
+For users of the `vim-plug`_ plugin manager, the vim Chapel support can be
+loaded as an unmanaged plugin. To do this, add the following line to the
+``.vimrc``, between the ``call plug#begin()`` and ``call plug#end()`` lines:
+
+.. code-block:: vim
+
+    Plug '~/path/to/chapel/highlight/vim'
+
+Where ``~/path/to/chapel`` is the path to the Chapel repository, starting from
+the home directory. Chapel support should work once these changes are made
+without any calls to ``:PlugInstall``, which is required for managed plugins.
+
+Note that vim-plug uses the ``~`` character to identify that a plugin is a local
+file, so the Chapel repository, or at least the ``highlight/vim/`` directory,
+must be somewhere under the home directory.
+
+.. _vim-plug: https://github.com/junegunn/vim-plug
+
+.. _vim-lsp-chpl:
+
+Language Server Support
+~~~~~~~~~~~~~~~~~~~~~~~
+
+The use the :ref:`Chapel language server <readme-chpl-language-server>` and :ref:`readme-chplcheck`, you will need to have an extension that supports the Language Server Protocol (LSP). There are a few options for this, these instructions will use the `vim-lsp <https://github.com/prabirshrestha/vim-lsp>`_ extension with `vim-plug`_.
+
+Using vim-lsp:
+^^^^^^^^^^^^^^
+
+Note that these instructions assume you have already setup Vim with Chapel syntax highlighting as described in the previous section.
+
+1. Assuming you have already installed the `vim-plug`_ plugin manager, you can add the following lines to your ``.vimrc`` file, between the between the ``call plug#begin()`` and ``call plug#end()`` lines:
+
+   .. code-block:: vim
+
+      Plug 'prabirshrestha/vim-lsp'
+
+2. After the ``call plug#end()`` line, add the following configuration:
+
+   .. code-block:: vim
+
+      if executable('chpl-language-server')
+          au User lsp_setup call lsp#register_server({
+              \ 'name': 'cls',
+              \ 'cmd': {server_info->['chpl-language-server', '--chplcheck']},
+              \ 'allowlist': ['chpl'],
+              \ })
+      endif
+
+      function! s:on_lsp_buffer_enabled() abort
+          setlocal omnifunc=lsp#complete
+          setlocal signcolumn=yes
+          if exists('+tagfunc') | setlocal tagfunc=lsp#tagfunc | endif
+          nmap <buffer> gd <plug>(lsp-definition)
+          nmap <buffer> gs <plug>(lsp-document-symbol-search)
+          nmap <buffer> gS <plug>(lsp-workspace-symbol-search)
+          nmap <buffer> gr <plug>(lsp-references)
+          nmap <buffer> <leader>rn <plug>(lsp-rename)
+          nmap <buffer> [g <plug>(lsp-previous-diagnostic)
+          nmap <buffer> ]g <plug>(lsp-next-diagnostic)
+          nmap <buffer> K <plug>(lsp-hover)
+      endfunction
```

Comment:
I suggest adding a comment that this provides some keybindings, because this is weird out of nowhere, and would alienate me as a vim user with a non-empty config